### PR TITLE
feat(react-wallet): Link wallet data to app store

### PR DESCRIPTION
**Pull request by samsiegart**
This gets all the dapps, purses, etc. printing in the console and updating when the notifiers update. For testing, I would like to paste some mock JSON data into a test file, but was having trouble getting it from the console because JSON.parse couldn't handle BigInt. I can follow up with that in another PR, or dig into that more depending on reviewer feedback.

### CODE REVIEW

**Comment by michaelfig** [comment]
> For testing, I would like to paste some mock JSON data into a test file, but was having trouble getting it from the console because JSON.parse couldn't handle BigInt.

You can post the mock data into a file just as an evaluable JS string (like `'{ a: 1n, b: 3n }'`), then `eval(s)` to get the actual JS data structure.  Don't do this for prod, but for testing it's fine to `eval` the string.

For bonus points under SES, use `c = new Compartment(); mystruct = c.evaluate(evaluableJsString);`

File: packages/dapp-svelte-wallet/_agstate/agoric-servers/package.json [hunk 1]
```diff
@@ -12,6 +12,6 @@
   "author": "Agoric",
   "license": "Apache-2.0",
   "dependencies": {
-    "@agoric/vats": "^0.1.0"
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/_agstate/agoric-servers/package.json]: Please don't use `agoric install` when you're in the SDK, just do a top-level `yarn install` to get the modules to install with proper version numbers.

So, please revert this `package.json` and other `package.json`s, only add dependencies (and their proper versions) when you need.

File: packages/dapp-svelte-wallet/api/package.json [hunk 2]
```diff
@@ -15,20 +15,20 @@
     "lint:eslint": "eslint '**/*.js'"
   },
   "devDependencies": {
-    "@agoric/bundle-source": "^2.0.0",
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/api/package.json]: Same.

File: packages/dapp-svelte-wallet/package.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
     "build": "exit 0"
   },
   "dependencies": {
-    "agoric": "^0.13.19",
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/package.json]: Same.

File: packages/dapp-svelte-wallet/react-ui/src/contexts/Application.jsx [hunk 4]
```diff
@@ -17,14 +24,87 @@ export const withApplicationContext = (Component, mapContextToProps) => {
...
+  if (a > b) {
+    return 1;
+  }
+  return 0;
+};
+
+const kv = (keyObj, val) => {
+  const key = Object.values(keyObj)[0];
+  const text = Array.isArray(key) ? key.join('.') : key;
+  return { ...val, ...keyObj, id: text, text, value: val };
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/react-ui/src/contexts/Application.jsx]: Note that the "id" can be taken from the newly-implemented `meta` property.  All these notifiable data things should have it, so you could expose it.

File: packages/dapp-svelte-wallet/ui/package.json [hunk 5]
```diff
@@ -16,9 +16,9 @@
     "test:xs": "exit 0"
   },
   "devDependencies": {
-    "@agoric/assert": "^0.3.14",
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/ui/package.json]: Same here.

**Review by michaelfig — COMMENTED** [review]
Please revert the package.json changes and consider the `eval(s)` way of generating test data, otherwise this is a great PR.

I'll leave this as just a comment and not an approval... feel free to ping me when you're R4R again.

**Comment by samsiegart** [comment]
> You can post the mock data into a file

I was having trouble getting the data from the console. I tried doing "copy object" like below, but it just shows [Object object] when I paste it:

image

I also tried JSON.stringify but get the BigInt error:
image

I'll probably have to type up the data manually. For component tests this is what should be done anyway, skipping the parsing step. For testing the parsing in contexts/Application.jsx I think that'd be better done in an integration test anyway but I can do so if necessary.

File: packages/dapp-svelte-wallet/react-ui/src/contexts/Application.jsx (hunk 4 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/contexts/Application.jsx]: Looks `meta` doesn't contain `id`:
image

I'll definitely include the timestamps when I create the request list though.

Also seems like purses don't have a `meta` field.

File: packages/dapp-svelte-wallet/_agstate/agoric-servers/package.json (hunk 1 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/_agstate/agoric-servers/package.json]: Whoops, fixed here and everywhere.

File: packages/dapp-svelte-wallet/react-ui/src/contexts/Application.jsx (hunk 4 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/contexts/Application.jsx]: Just changed to use `sequence` as the id where applicable as discussed.

File: packages/dapp-svelte-wallet/api/src/types.js [hunk 6]
```diff
@@ -98,7 +98,7 @@
 
 /**
  * @typedef {Object} RecordMetadata
- * @property {number} sequence
+ * @property {number} id
  * A monotonically increasing number to allow total ordering between records.
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/api/src/types.js]: Sorry, should have caught this before.

```suggestion
 * Identifies a particular Record in the context of the wallet backend.  This id is stable and unique for a wallet backend (even across different record types).
```

**Review by michaelfig — APPROVED** [review]
Please clean up the `id` type comment... your own words would be fine.

Otherwise, LGTM!

File: packages/dapp-svelte-wallet/api/src/types.js (hunk 6 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/api/src/types.js]: Fixed

**Pull request merged**